src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorFileChangeDetectorManager.cs

```diff
@@ -41,6 +43,23 @@ public async Task InitializedAsync(ILanguageServer languageServer)
                 var cancellationToken = CancellationToken.None;
                 await fileChangeDetector.StartAsync(workspaceDirectory, cancellationToken);
             }
+
+            lock (_disposeLock)
+            {
+                if (_disposed)
+                {
+                    // Got disposed while starting our file change detectors. We need to re-stop our change detectors.
```

Comment:
Is this in the interest of being defensive, or is this actually expected to happen frequently?

Reply:
It can definitely happen. Imagine our server is in the midst of booting up and then the user closes the solution. LSP project will issue a shutdown request in what could be the middle of our file change initialization.